pkg/monitortests/node/legacynodemonitortests/kubelet.go

```diff
@@ -245,11 +248,17 @@ func testContainerFailures(events monitorapi.Intervals) []*junitapi.JUnitTestCas
 
 	var excessiveExits []string
 	maxRestartCount := 3
+
+	isUpgrade := platformidentification.DidUpgradeHappenDuringCollection(events, time.Time{}, time.Time{})
+
+	clusterDataPlatform, _ := platformidentification.BuildClusterData(context.Background(), adminRestConfig)
+
+	exclusions := Exclusion{upgradeJob: isUpgrade, clusterData: clusterDataPlatform}
 	for locator, messages := range containerExits {
 		if len(messages) > 0 {
 			messageSet := sets.NewString(messages...)
-			// Blanket fail for restarts over 3
-			if len(messages) > maxRestartCount {
+			// Blanket fail for restarts over maxRestartCount
+			if !isThisContainerRestartExcluded(locator, exclusions) && len(messages) > maxRestartCount {
```

Comment:
How about we start failing (rather than flaking) for maxRestartCount=3.  And flake a *different* test name for maxRestartCount=2.